src/org/zaproxy/zap/extension/pscanrulesAlpha/CookieSameSiteScanner.java

```diff
@@ -0,0 +1,134 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.Vector;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.htmlparser.jericho.Source;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+
+public class CookieSameSiteScanner extends PluginPassiveScanner {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "pscanalpha.cookiesamesitescanner.";
+	private static final int PLUGIN_ID = 10054;
+
+	private static final String SAME_SITE_COOKIE_ATTRIBUTE = "SameSite";
+	private static final String SAME_SITE_COOKIE_VALUE_STRICT = "Strict";
+	private static final String SAME_SITE_COOKIE_VALUE_LAX = "Lax";
+	
+	private PassiveScanThread parent = null;
+
+	@Override
+	public void setParent (PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage msg, int id) {
+		// Ignore
+	}
+
+	@Override
+	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		checkCookies(msg, id, HttpHeader.SET_COOKIE);
+		checkCookies(msg, id, HttpHeader.SET_COOKIE2);
+	}
+	
+	private void checkCookies(HttpMessage msg, int id, String cookieHeader) {
+		Vector<String> cookies = msg.getResponseHeader().getHeaders(cookieHeader);
+
+		if (cookies == null) {
+			return;
+		}
+		for (String cookie : cookies) {
+			String sameSiteVal = SetCookieUtils.getAttributeValue(cookie, SAME_SITE_COOKIE_ATTRIBUTE);
+			String name = SetCookieUtils.getCookieName(cookie);
+			if (sameSiteVal == null) {
+				// Its missing
+				this.raiseAlert(msg, id, name, this.getDescription());
+			} else if (! (sameSiteVal.equalsIgnoreCase(SAME_SITE_COOKIE_VALUE_STRICT) ||
+					sameSiteVal.equalsIgnoreCase(SAME_SITE_COOKIE_VALUE_LAX))) {
+				// Its present but with an illegal value
+				this.raiseAlert(msg, id, name,  
+						Constant.messages.getString(MESSAGE_PREFIX + "badval"));
+			}
+		}
+		
+	}
+	
+	private void raiseAlert(HttpMessage msg, int id, String name, String description) {
+		String evidence = name;
+		// Try to find the exact Set-Cookie string
+		Pattern pattern = Pattern.compile("Set-Cookie.*" + name, Pattern.CASE_INSENSITIVE);
+		Matcher matcher = pattern.matcher(msg.getResponseHeader().toString());
+		if (matcher.find()) {
+			evidence = matcher.group();
```

Comment:
Couldn't there be multiple matches at this point? So perhaps you want `group(1)`?